crates/sui-types/src/move_package.rs

```diff
@@ -559,6 +574,9 @@ fn build_linkage_table<'p>(
     let mut dep_linkage_tables = vec![];
 
     for transitive_dep in transitive_dependencies.into_iter() {
+        // TODO: original_package_id will deserialized a module but only for the purpose of
```

Comment:
Hmm, technically yes, but I don't think practically it could make a difference, because we would have needed to write out these packages to the store at a greater version, and presumably if we don't ever lower the max supported version, that means this node would have had to successfully execute that transaction.

Will leave it with you if you want to dot this particular `i`, but I don't think it's strictly necessary -- if we're worried about this, I think my vote is to implement @tzakian's idea of caching a package's original ID in `MovePackage` directly.

Reply:
I'd say let's not worry about if for now then